JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesRDBClientWithReducedTables.java

```diff
@@ -228,49 +367,53 @@ public void initTimeSeriesTable(List<String> dataIRIList, List<Class<?>> dataCla
     @Override
     public void addTimeSeriesData(List<TimeSeries<T>> tsList, Connection conn) {
 
+        // Check if central database lookup table exists
+        if (!checkCentralTableExists(conn)) {
+            throw new JPSRuntimeException(
+                    exceptionPrefix + "Central RDB lookup table has not been initialised yet");
+        }
+
         // Initialise connection and set jOOQ DSL context
         DSLContext context = DSL.using(conn, DIALECT);
 
-        // Loop over all time series in list
-        for (TimeSeries<T> ts : tsList) {
-
-            List<String> dataIRI = ts.getDataIRIs();
+        // All database interactions in try-block to ensure closure of connection
 
-            // All database interactions in try-block to ensure closure of connection
-            try {
+        try {
+            Statement statement = conn.createStatement();
 
-                // Check if central database lookup table exists
-                if (!checkCentralTableExists(conn)) {
-                    throw new JPSRuntimeException(
-                            exceptionPrefix + "Central RDB lookup table has not been initialised yet");
-                }
+            List<String> listStmt = tsList.parallelStream().map(ts -> {
+                List<String> dataIRI = ts.getDataIRIs();
 
                 // Ensure that all provided dataIRIs/columns are located in the same RDB table
                 // (throws Exception if not)
                 checkDataIsInSameTable(dataIRI, context);
 
                 String tsIri = getTimeSeriesIRI(dataIRI.get(0), context);
                 // Assign column name for each dataIRI; name for time column is fixed
-                Map<String, String> dataColumnNames = new HashMap<>();
-                for (String s : dataIRI) {
-                    dataColumnNames.put(s, getColumnName(s, context));
-                }
+                Map<String, String> dataColumnNames = bulkGetColumnName(dataIRI, context);
 
                 // Append time series data to time series table
                 // if a row with the time value exists, that row will be updated instead of
                 // creating a new row
-                populateTimeSeriesTable(ts, dataColumnNames, tsIri, conn);
-            } catch (JPSRuntimeException e) {
-                // Re-throw JPSRuntimeExceptions
-                throw e;
-            } catch (Exception e) {
-                LOGGER.error(e.getMessage());
-                // Throw all exceptions incurred by jooq (i.e. by SQL interactions with
-                // database) as JPSRuntimeException with respective message
-                throw new JPSRuntimeException(exceptionPrefix + "Error while executing SQL command", e);
+                return populateTimeSeriesTable(ts, dataIRI, dataColumnNames, tsIri, conn);
```

Comment:
I don't see any issues with querying in parallel, but I'm a bit cautious about adding data in parallel. If you have tested this extensively then I'm fine.